compiler/rustc_data_structures/src/graph/mod.rs

```diff
@@ -20,55 +20,40 @@ pub trait WithNumEdges: DirectedGraph {
     fn num_edges(&self) -> usize;
 }
 
-pub trait WithSuccessors: DirectedGraph
-where
-    Self: for<'graph> GraphSuccessors<'graph, Item = <Self as DirectedGraph>::Node>,
-{
-    fn successors(&self, node: Self::Node) -> <Self as GraphSuccessors<'_>>::Iter;
+pub trait Successors: DirectedGraph {
```

Comment:
I wonder if it would make sense to use RPITIT in both `Successors` and `Predecessors` instead of an explicit associated type? Or ATPIT / ITIAT if RPITIT doesn't work out.

Might clean up things even more. Only do this if it's a clear improvement though.